src/beaconDecoder.py

```diff
@@ -18,8 +27,29 @@ def run(self):
         beacon_port = 1
         payload_offset = 16
         packet_num_offset = 4
-        decoded_data = base64.b64decode(self.rawdata)
-        decoded_data = bytearray(decoded_data)
+
+        # Check if socket is still open
+        try:
+            self.s.send(bytes("You good bro?", 'utf-8'))
```

Comment:
Are you sure this is a good idea? I thought there were calls like stat(2) to non-invasively check socket status.

Reply:
Besides which, self.parser should have warned you on receive if there was a problem

Reply:
I couldn't figure out how to do it non-invasively

Reply:
Figured it out now - but moved the changes to another branch